core/src/main/java/org/mule/endpoint/MuleEndpointURI.java

```diff
@@ -494,37 +498,70 @@ public String getFilterAddress()
 
     public String getUser()
     {
-        if (StringUtils.isNotBlank(userInfo))
+        if (user == null)
         {
-            int i = userInfo.indexOf(':');
-            if (i == -1)
-            {
-                return userInfo;
-            }
-            else
+            user = getUserInfoDataUsing(new DataExtractor()
             {
-                return userInfo.substring(0, i);
-            }
+                @Override
+                public String extract(String source)
+                {
+                    int i = source.indexOf(':');
+                    if (i == -1)
+                    {
+                        return source;
+                    }
+                    else
+                    {
+                        return source.substring(0, i);
+                    }
+                }
+            });
         }
-        return null;
+        return user;
     }
 
-    public String getResponseTransformers()
+    public String getPassword()
```

Comment:
Add javadoc to the implemented method in the interface to indicate that the returned value is url-decoded.
Also add @Override